app/models/template.rb

```diff
@@ -99,14 +102,23 @@ def import_without_save(text, options = {})
     self
   end
 
+  def valid_params?(name, metadata)
+    if !metadata["kind"].nil? && (metadata["snippet"] == true)
+      raise Foreman::Exception.new(N_('snippet and template kind don\'t match'))
```

Comment:
That is not very informative message, it doesn't tell me much about what exactly I should fix, does it?

Reply:
sure, do you have a better suggestion? :)